DataFormats/CTPPSDigi/interface/CTPPSPixelDigi.h

```diff
@@ -0,0 +1,69 @@
+#ifndef CTPPS_CTPPSDigi_CTPPSPixelDigi_h
+#define CTPPS_CTPPSDigi_CTPPSPixelDigi_h
+/**
+ * Persistent digi for CTPPS Pixels.
+ * Author: F.Ferro ferro@ge.infn.it
+ */
+
+#include "FWCore/Utilities/interface/Exception.h"
+
+class CTPPSPixelDigi {
+public:
+
+
+  CTPPSPixelDigi( int packed_value) : theData(packed_value) {}
+
+  CTPPSPixelDigi( int row, int col, int adc) {
+    init( row, col, adc);
+  }
+
+  CTPPSPixelDigi( int chan, int adc) {
+    std::pair<int,int> rc = channelToPixel(chan);
+    init( rc.first, rc.second, adc);
+  }
+
+  CTPPSPixelDigi() : theData(0)  {}
+
+  /// Access to digi information
+  int row() const     {return (theData >> row_shift) & row_mask;}
+  int column() const  {return (theData >> column_shift) & column_mask;} 
+  unsigned short adc() const  {return (theData >> adc_shift) & adc_mask;}
+  uint32_t packedData() const {return theData;}
+
+  static std::pair<int,int> channelToPixel( int ch) {
+    int row = ( ch >> column_width) & row_mask;
```

Comment:
then I guess I don't follow the variable names and now the values correspond to the example you gave. 

Naively I expected
row_shift : the first bit used to store the row information
row_mask : a 1 for each bit used to store the row (you defined this to be after the shift, so it just corresponds to row_width "1"s. 

column_shift, column_mask : the same but for columns
adc_shift, add_mask : the same but for adc values

the definitions below would correspond to 
aaaaaaaaaaccccccccccccrrrrrrrrrrrr
(a = adc bit, c = column bit, r = row bit)

so then 

row = (channel  >> row_shift ) & row_mask

but that doesn't match your example or your code above